File: codeforces/cf520A_pangram.rs
// https://codeforces.com/problemset/problem/520/A
// basic string manipulation, set
use std::io;
use std::collections::HashSet;

fn main() {
    io::stdin()
        .read_line(&mut String::new())
        .unwrap();

    let mut line = String::new();

    io::stdin()
        .read_line(&mut line)
        .unwrap();

    let mut line: String = line
        .trim()
        .to_string();

    line.make_ascii_lowercase();

    let mut letters = HashSet::new();

    for ch in line.chars() {
        letters.insert(ch);
    }

    let alphabet = "abcdefghijklmnopqrstuvwxyz";

    let mut flag = true;

    for ch in alphabet.chars() {
        match letters.get(&ch) {
            Some(_) => continue,
            None => {
                flag = false;
                break;
            },
        }
    }

    if flag {
        println!("YES");
    } else  {
        println!("NO");
    }
}

File: codeforces/cf520A_pangram.cpp
// https://codeforces.com/problemset/problem/520/A

#include<iostream>
#include<algorithm>
#include<string>

using namespace std;

int main() {
    int n;
    string phrase;
    string alphabet {"abcdefghijklmnopqrstuvwxyz"};

    cin >> n >> phrase;

    if (phrase.length() < alphabet.length()) {
        cout << "NO";
        return 0;
    }

    transform(phrase.begin(), phrase.end(), phrase.begin(), ::tolower);

    bool flag = true;
    for (auto letter : alphabet) {
        if (phrase.find(letter) == string::npos) {  // letter not found.
            flag = false;
            break;
        }
    }

    if (flag) cout << "YES";
    else cout << "NO";

    return 0;
}
